DM-50191: Add GLOB() function to user expression language (#1189)
lsst/daf_butler: doc/lsst.daf.butler/queries.rst

Function call syntax is similar to other languages, expression for call
consists of an identifier followed by zero or more comma-separated arguments
enclosed in parentheses (e.g. ``func(1, 2, 3)``). An argument to a function
can be any expression. Names of the functions are not case-sensitive.

> Presently there only few functions implemented in query language:

Reviewer: ```suggestion
Presently the following functions are implemented in the query language:
```
Does make me wonder whether we should add the IVOA `POS` syntax since sphgeom already supports that.
